### PR TITLE
Fix(migration): Unblock rollback of failed migrations

### PR DESCRIPTION
**Pull request by manadart**
This fixes indefinite blocking of failed model migration roll-backs.

Attempting to delete the read-only model data from the model database was stone cold blocking when attempting to acquire the transaction runner in `StateBase.DB()`. This will be investigated further, but this step is not actually required.

The clean-up done by `DeleteDB` in the `dbaccessor` worker covers this removal, we just have to attempt it in all cases.

## QA steps

- Bootstrap (src) and add a model.
- Bootstrap (dst) and migrate the src model to this controller.
- Check that the failed migration is rolled back.
- Check the logs on the dst controller:
```
machine-0: 08:45:26 INFO juju.apiserver running operation: import leases
machine-0: 08:45:26 INFO juju.apiserver running operation: import external controllers
machine-0: 08:45:26 INFO juju.apiserver running operation: import cloud credentials
machine-0: 08:45:26 INFO juju.apiserver running operation: import model
machine-0: 08:45:26 ERROR juju.apiserver import failed: execute operation import model: importing model "work" with id "234ec633-f8d6-4893-8adf-740016f08d10" during migration: cannot import model with id "234ec633-f8d6-4893-8adf-740016f08d10", agent version cannot be zero: agent version not supported
machine-0: 08:45:26 INFO juju.apiserver rolling back operation: import model
machine-0: 08:45:26 INFO juju.worker.dbaccessor deleting database for namespace "234ec633-f8d6-4893-8adf-740016f08d10"
machine-0: 08:45:26 INFO juju.apiserver rolling back operation: import cloud credentials
machine-0: 08:45:26 INFO juju.apiserver rolling back operation: import external controllers
machine-0: 08:45:26 INFO juju.apiserver rolling back operation: import leases
machine-0: 08:45:26 DEBUG juju.worker.state model "234ec633-f8d6-4893-8adf-740016f08d10" removed from state
```

## Documentation changes

None.

## Links

**Jira card:** [JUJU-6147](https://warthogs.atlassian.net/browse/JUJU-6147)

[JUJU-6147]: https://warthogs.atlassian.net/browse/JUJU-6147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: domain/model/modelmigration/import.go [hunk 1]
```diff
@@ -223,23 +223,15 @@ func (i importOperation) Execute(ctx context.Context, model description.Model) e
...
-func (i importOperation) Rollback(ctx context.Context, model description.Model) error {
-	// Attempt to rollback the model database if it was created.
+func (i *importOperation) Rollback(ctx context.Context, model description.Model) error {
+	// Attempt to roll back the model database if it was created.
 	modelName, modelID, err := i.getModelNameAndID(model)
 	if err != nil {
 		return fmt.Errorf("rollback of model during migration %w", errors.NotValid)
 	}
 
-	// If the read only model isn't found, we can simply ignore the error.
```
Comment (manadart) [inline domain/model/modelmigration/import.go:235]: This call is the one that was blocking everything, but it should be covered by the `DeleteModel` call below.

File: domain/model/state/modelstate.go [hunk 2]
```diff
@@ -50,6 +49,7 @@ func (s *ModelState) Create(ctx context.Context, args model.ReadOnlyModelCreatio
 
 // Delete deletes a model.
 func (s *ModelState) Delete(ctx context.Context, uuid coremodel.UUID) error {
+	// TODO (manadart 2024-06-06): Check why we block here prior to PR 17472.
```
Comment (manadart) [inline domain/model/state/modelstate.go:52]: This is where we actually block. We'll remove this comment post-resolution.

**Review by jack-w-shaw — APPROVED** [review]
QA passes, LGTM

File: domain/model/service/service.go [hunk 3]
```diff
@@ -382,7 +382,7 @@ func (s *Service) DeleteModel(
 
 	// Delete common items from the model. This helps to ensure that the
 	// model is cleaned up correctly.
-	if err := s.st.Delete(ctx, uuid); err != nil {
+	if err := s.st.Delete(ctx, uuid); err != nil && !errors.Is(err, modelerrors.NotFound) {
```
Comment (manadart) [inline domain/model/service/service.go:385]: This is so we always attempts the DB deletion below, even if we don't think the model is there.

**Pull request merged**